### PR TITLE
ECC: ECDSA at GP level

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by pascal-brand-st-dev** [comment]
@cedric-chaumont-st any comment?

File: core/tee/tee_svc_cryp.c [hunk 1]
```diff
@@ -1844,6 +1844,15 @@ static TEE_Result tee_svc_cryp_check_key_type(const struct tee_obj *o,
...
 		break;
+	case TEE_MAIN_ALGO_ECDSA:
+		if (mode == TEE_MODE_VERIFY)
+			req_key_type = TEE_TYPE_ECDSA_PUBLIC_KEY;
+		else
+			req_key_type = TEE_TYPE_ECDSA_KEYPAIR;
+		break;
+	case TEE_MAIN_ALGO_ECDH:
+		req_key_type = TEE_TYPE_ECDH_KEYPAIR;
+		break;
```
Comment (cedric-chaumont-st-dev) [inline core/tee/tee_svc_cryp.c]: Should  TEE_MAIN_ALGO_ECDH/TEE_MAIN_ALGO_DSA cases be updated (same as TEE_MAIN_ALGO_ECDSA) ?

Comment (pascal-brand-st-dev) [inline core/tee/tee_svc_cryp.c]: I do not think so. We only check here the key type, not all the mode.

However, I think you pointed something not supported right now: VERIFY / ENCRYPT must also work on keypair. In that case, only the public key of the keypair is taking into account. I'll submit this in another patch.

**Comment by pascal-brand-st-dev** [comment]
Fix ECC self test - Message to encrypt must be the size of a SHA1 digest

xtest has been updated to include ECDSA sign & verify, at https://github.com/OP-TEE/optee_test/pull/8

**Comment by cedric-chaumont-st-dev** [comment]
Reviewed-by: Cedric Chaumont <cedric.chaumont@linaro.org>

**Pull request merged**